security/pkg/server/ca/server.go

```diff
@@ -165,7 +165,7 @@ func Authenticate(ctx context.Context, auth []security.Authenticator) *security.
 	// TODO: apply different authenticators in specific order / according to configuration.
 	var errMsg string
 	for id, authn := range auth {
-		u, err := authn.Authenticate(ctx)
+		u, err := authn.Authenticate(security.NewAuthContext(ctx))
```

Comment:
this differs from the XDS logic in ways that seem like they would break things. In the XDS one we share an AuthContext, here we create one per authenticator.

Can/should we refactor them to use a shared impl?

Reply:
Added AuthenticationManager take orchestrates Authenticators. PTAL.